### PR TITLE
Add documentation on how to analyze TCJA expiration

### PR DESCRIPTION
**Pull request by martinholmer**
This pull request adds only documentation; it includes no substantive changes to the tax calculation logic.

### CODE REVIEW

File: docs/usage/ext.json
Comment (jdebacker) [inline docs/usage/ext.json:1]: Should this file go in the `taxcalc/reforms/` directory?

File: docs/usage/tcja_after_2025.md [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+you have named your reform file `x.json` and that you are using one
+of the compatible input datasets describe above.  The examples will
+call the input dataset `z.csv`.
+
+To analyze your reform relative to current-law policy (which means
+temporary TCJA provisions have expired beginning in 2026), you would
+execute this command:
+
+```
+tc z.csv 2026 --exact --tables --reform x.json
```
Comment (jdebacker) [inline docs/usage/tcja_after_2025.md:69]: This instructions are very helpful, but to ease the burden on those replicating them, and to provide a concrete example, should their be a full example in the "recipes" using the current policy baseline?

Comment (martinholmer) [reply docs/usage/tcja_after_2025.md:69]: @jdebacker said:
> This instructions are very helpful, but to ease the burden on those replicating them, and to provide a concrete example, should their be a full example in the "recipes" using the current policy baseline?

I don't have the time to do that and I don't have any experience with Jupyter files, which is what the Recipes have been converted to.   If you want a recipe, ask someone with the necessary skills.

File: docs/usage/ext.json
Comment (martinholmer) [reply docs/usage/ext.json:1]: @jdebacker said:
> Should this file go in the `taxcalc/reforms/` directory?

Maybe, but it seemed to me that each file in `reforms/` is tested.   I don't have the time to maintain test results for the `ext.json` reform.

**Comment by martinholmer** [comment]
@jdebacker asked this question after reviewing PR #2734:
> These instructions are very helpful, but to ease the burden on those replicating them, and to provide a concrete example, should their be a full example in the "recipes" using the current policy baseline?

I considered that but felt it was not the best option for three reasons:
* All the "recipes" demonstrate how to use the **Python API** to Tax-Calculator, not the **CLI** to Tax-Calculator.   So, putting it there would be like adding an orange to a barrel of apples.
* Putting this CLI TCJA document in the Recipes section would make it invisible in the high-level documentation table of contents.
* The recipes are now Jupyter files and I don't have the time or skills to develop that kind of documentation.

**Comment by martinholmer** [comment]
@jdebacker asked this question in his review of PR #2734:
> Should [`ext.json`] go in the taxcalc/reforms/ directory?

I considered that but rejected it as suboptimal.
The main problem is that putting `ext.json` in the `taxcalc/reforms/` directory means that it must be tested by the `taxcalc/tests/test_reforms.py` script.  That would be acceptable if it weren't for the fact that the `ext.json` reform and its tests results will be (unlike the other reforms) changing several times over the course of the next year.   The reasons for these future changes are explained in the `tcja_after_2025.md` document.   My understanding of the files in the `taxcalc/reforms/` directory is that they are historical, and therefore, not in flux.   So, there would be a considerable extra time burden in maintaining `ext.json` if it were located in the `taxcalc/reforms/` directory.

**Comment by jdebacker** [comment]
@martinholmer writes:
> I considered that but rejected it as suboptimal.
The main problem is that putting ext.json in the taxcalc/reforms/ directory means that it must be tested by the taxcalc/tests/test_reforms.py script. That would be acceptable if it weren't for the fact that the ext.json reform and its tests results will be (unlike the other reforms) changing several times over the course of the next year. The reasons for these future changes are explained in the tcja_after_2025.md document. My understanding of the files in the taxcalc/reforms/ directory is that they are historical, and therefore, not in flux. So, there would be a considerable extra time burden in maintaining ext.json if it were located in the taxcalc/reforms/ directory.

Martin, updates to `ext.json` should themselves be tested.  That is good practice.  For example, what if a syntax error is introduced in the JSON file? We'd want to be able to catch that.  The tests do that.

In addition, the `reform` directory is the place users are accustom to looking for JSON files like this -- not the `docs` folder. I could see the case for having an alternative current policy baseline assumption like this in the `taxcalc` directory (alongside the current law baseline), but I don't see how it makes sense to put this file in the `docs/usage` directory.

Before merging, I would like to see this added to the `reforms` folder and tested to help reduce the likelihood errors are introduced in the future.  Thank you!

**Comment by martinholmer** [comment]
@jdebacker said:
> Before merging [PR #2734], I would like to see [`ext.json`] added to the reforms folder and tested to help reduce the likelihood errors are introduced in the future. Thank you!

OK.  I have done what you asked.
But now I cannot do a `git push origin add-tcja-docs` to move my local branch to the GitHub PR.  Here is what I'm getting:

```
(taxcalc-dev) Tax-Calculator% git push origin add-tcja-docs
To https://github.com/martinholmer/Tax-Calculator.git
 ! [rejected]          add-tcja-docs -> add-tcja-docs (non-fast-forward)
error: failed to push some refs to 'https://github.com/martinholmer/Tax-Calculator.git'
hint: Updates were rejected because the tip of your current branch is behind
hint: its remote counterpart. Integrate the remote changes (e.g.
hint: 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.

(taxcalc-dev) Tax-Calculator% git branch     
* add-tcja-docs
  master

(taxcalc-dev) Tax-Calculator% git pull origin
You asked to pull from the remote 'origin', but did not specify
a branch. Because this is not the default configured remote
for your current branch, you must specify a branch on the command line.

(taxcalc-dev) Tax-Calculator% git pull origin add-tcja-docs
From https://github.com/martinholmer/Tax-Calculator
 * branch              add-tcja-docs -> FETCH_HEAD
hint: You have divergent branches and need to specify how to reconcile them.
hint: You can do so by running one of the following commands sometime before
hint: your next pull:
hint: 
hint:   git config pull.rebase false  # merge
hint:   git config pull.rebase true   # rebase
hint:   git config pull.ff only       # fast-forward only
hint: 
hint: You can replace "git config" with "git config --global" to set a default
hint: preference for all repositories. You can also pass --rebase, --no-rebase,
hint: or --ff-only on the command line to override the configured default per
hint: invocation.
fatal: Need to specify how to reconcile divergent branches.
```

I have no idea what `git` is saying here.   I'm doing the same thing as I always do.
What is different now?   Something related to the new Codecov token?   Or something else?

**Comment by jdebacker** [comment]
Thank you @martinholmer for this work. Nice addition to allow for a current policy baseline.

**Pull request merged**